### PR TITLE
Onboarding task9

### PR DESCRIPTION
**Pull request by thalyscarrara**
## Description

dependencie injection with Hilt

## Related Issues

[task09-onboarding](https://github.com/profusion/quickstart-android-jetpack-graphql/issues/227)

## Visual reference

https://user-images.githubusercontent.com/53837410/193362335-084bd4c3-6279-4b46-9622-d14e2add2d71.mp4

### CODE REVIEW

File: app/src/main/java/com/example/android/hilt/di/DatabaseModule.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            AppDatabase::class.java,
+            "logging.db"
+        ).build()
+    }
+
+    @Provides
+    fun provideLogDao(database: AppDatabase): LogDao {
+        return database.logDao()
+    }
+}
```
Comment (mari1912) [inline app/src/main/java/com/example/android/hilt/di/DatabaseModule.kt]: Don't forget to add the linter `ktlint` to Android Studio. It should be pointing out that you should add a new line

File: app/src/main/java/com/example/android/hilt/navigator/AppNavigatorImpl.kt [hunk 2]
```diff
@@ -20,11 +20,12 @@ import androidx.fragment.app.FragmentActivity
 import com.example.android.hilt.R
 import com.example.android.hilt.ui.ButtonsFragment
 import com.example.android.hilt.ui.LogsFragment
+import javax.inject.Inject
 
 /**
  * Navigator implementation.
  */
```
Comment (mari1912) [inline app/src/main/java/com/example/android/hilt/navigator/AppNavigatorImpl.kt]: I know that you're following the codelab, but a good exercise it's to look at all the comments and think if it's really necessary to keep it or if the code is already self-explained

File: app/src/main/java/com/example/android/hilt/di/NavigationModule.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ActivityComponent
+
+@InstallIn(ActivityComponent::class)
+@Module
+abstract class NavigationModule {
+
+    @Binds
+    abstract fun bindNavigator(impl: AppNavigatorImpl): AppNavigator
+}
```
Comment (mari1912) [inline app/src/main/java/com/example/android/hilt/di/NavigationModule.kt]: new line

**Review by mari1912 — APPROVED** [review]
pre-approving

**Review by joaobertacchi — APPROVED** [review]

**Comment by joaobertacchi** [comment]
For documenting purposes, if we use this codelab in the future, the following changes are required to make it work on Apple M1:
```
diff --git a/app/build.gradle b/app/build.gradle
index ebe67a2..d105c49 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -56,8 +56,8 @@ dependencies {
     implementation 'androidx.recyclerview:recyclerview:1.2.1'

     // Room
-    implementation "androidx.room:room-runtime:2.3.0"
-    kapt "androidx.room:room-compiler:2.3.0"
+    implementation "androidx.room:room-runtime:2.4.3"
+    kapt "androidx.room:room-compiler:2.4.3"

     // Testing dependencies
     androidTestImplementation "junit:junit:4.13.2"
diff --git a/build.gradle b/build.gradle
index 7585ed9..7d9bb93 100644
--- a/build.gradle
+++ b/build.gradle
@@ -15,7 +15,7 @@
  */

 buildscript {
-    ext.kotlin_version = '1.5.31'
-     ext.hilt_version = '2.40'
+    ext.kotlin_version = '1.7.10'
+     ext.hilt_version = '2.44'
     repositories {
         google()
```

**Pull request merged**